arch/rv32i/src/syscall.rs

```diff
@@ -52,32 +52,56 @@ impl SysCall {
 impl kernel::syscall::UserspaceKernelBoundary for SysCall {
     type StoredState = Riscv32iStoredState;
 
+    fn initial_process_app_brk_size(&self) -> usize {
+        // TOCK 1.X
+        //
+        // We do not need any memory to start with, but the 1.x Tock kernel
+        // allocates at least 3 kB to processes, and we need to ensure that
+        // happens as userspace may expect it.
+        3 * 1024
+
+        // TOCK 2.0
+        //
+        // The RV32I UKB implementation does not use process memory for any
+        // context switch state. Therefore, we do not need any process-accessible
+        // memory to start with to successfully context switch to the process the
+        // first time.
+        //0
+    }
+
     unsafe fn initialize_process(
         &self,
-        stack_pointer: *const usize,
-        _stack_size: usize,
+        accessible_memory_start: *const u8,
+        _app_brk: *const u8,
         state: &mut Self::StoredState,
-    ) -> Result<*const usize, ()> {
+    ) -> Result<(), ()> {
         // Need to clear the stored state when initializing.
         state.regs.iter_mut().for_each(|x| *x = 0);
         state.pc = 0;
         state.mcause = 0;
 
         // The first time the process runs we need to set the initial stack
         // pointer in the sp register.
-        state.regs[R_SP] = stack_pointer as u32;
+        //
+        // TOCK 1.X
+        state.regs[R_SP] = accessible_memory_start.add(3 * 1024) as u32;
```

Comment:
Until we switch to the tock 2.0 version, we technically need a check here that there is enough stack space for this to be safe I think. But maybe that doesn't matter for the 2.0-dev branch, because we should be switching to the 2.0 version before this will actually be used for anything